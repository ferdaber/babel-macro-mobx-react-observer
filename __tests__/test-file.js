const pluginTester = require("babel-plugin-tester").default;
const plugin = require("babel-plugin-macros");
const path = require("path");
const fs = require("fs");

pluginTester({
  plugin,
  snapshot: true,
  babelOptions: {
    filename: "fixture-1",
    presets: [
      [
        "@babel/preset-env",
        {
          modules: false,
          targets: {
            chrome: 80,
          },
        },
      ],
      "@babel/preset-react",
    ],
  },
  tests: [
    fs.readFileSync(path.resolve("__fixtures__", "fixture-1.js")).toString(),
  ],
});
