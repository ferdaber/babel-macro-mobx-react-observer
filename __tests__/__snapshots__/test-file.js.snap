// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`macros 1. macros: 1. macros 1`] = `

import useObserver from "./index.macro";
import * as Mobx from "mobx-react-lite";
import def, { someImport } from "mobx-react-lite";
import "mobx-react-lite";

function MyComponent(props) {
  useObserver();
  return <div />;
}

(function (props) {
  useObserver();
});

const a = someHoc((props, ref) => {
  useObserver(), (<div />);
});

const b = () => (useObserver(), (<div />));
const c = () => useObserver();

      ↓ ↓ ↓ ↓ ↓ ↓

import * as Mobx from "mobx-react-lite";
import def, { someImport, observer } from "mobx-react-lite";
import "mobx-react-lite";
const MyComponent = observer(function MyComponent(props) {
  return /*#__PURE__*/ React.createElement("div", null);
});
observer(function (props) {});
const a = someHoc(
  observer((props, ref) => {
    /*#__PURE__*/
    React.createElement("div", null);
  })
);
const b = observer(() => /*#__PURE__*/ React.createElement("div", null));
const c = observer(() => {});


`;
